File: app/src/main/java/com/schrondinger/quin/ui/main/fragment/TypeTwoFragment.java
package com.schrondinger.quin.ui.main.fragment;


import android.content.res.AssetManager;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.schrondinger.quin.R;
import com.schrondinger.quin.base.activity.ActivityInject;
import com.schrondinger.quin.base.activity.BaseFragment;

import java.io.IOException;
import java.io.InputStream;

import butterknife.BindView;
import io.reactivex.Flowable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.functions.Consumer;
import io.reactivex.functions.Function;
import io.reactivex.schedulers.Schedulers;

@ActivityInject(rootViewId = R.layout.fragment_type_two)
public class TypeTwoFragment extends BaseFragment {
    @BindView(R.id.testImage)
    ImageView mIv_testImage;

    private Bitmap getImageFromAssetsFile(String fileName)
    {
        Bitmap image = null;
        AssetManager am = getResources().getAssets();
        try
        {
            InputStream is = am.open(fileName);
            image = BitmapFactory.decodeStream(is);
            is.close();
        }
        catch (IOException e)
        {
            e.printStackTrace();
        }

        return image;

    }

    @Override
    public void initData() {
        super.initData();

        Flowable.just("goutouren.jpg"). // 被订阅的数据源
                subscribeOn(Schedulers.io()) // 指定 subscribe() 发生在 IO 线程
                .observeOn(AndroidSchedulers.mainThread()) // 指定 Subscriber 的回调发生在主线程
                .map(new Function<String, Bitmap>() {
                    @Override
                    public Bitmap apply(String s) {
                        return  getImageFromAssetsFile(s);
                    }
                }).subscribe(new Consumer<Bitmap>() {
            @Override
            public void accept(Bitmap bitmap) {
                mIv_testImage.setImageBitmap(bitmap);
            }
        });
    }

    @Override
    public void initView() {
        super.initView();
    }
}

File: app/src/main/java/com/schrondinger/quin/ui/main/fragment/TypeTwoFragment.kt
package com.schrondinger.quin.ui.main.fragment


import android.support.v4.app.Fragment
import android.support.v4.content.ContextCompat
import android.view.View
import com.schrondinger.quin.R
import com.schrondinger.quin.base.activity.ActivityInject
import com.schrondinger.quin.base.activity.BaseFragment
import com.schrondinger.quin.bean.common.CommMap
import com.schrondinger.quin.ui.main.MainActivity
import com.schrondinger.quin.ui.common.adaper.ContentFragmentPagerAdapter
import com.schrondinger.quin.ui.mine.fragment.MineOneFragment
import com.schrondinger.quin.ui.music.fragment.MusicFragment
import kotlinx.android.synthetic.main.fragment_type_two.*

@ActivityInject(rootViewId = R.layout.fragment_type_two)
class TypeTwoFragment : BaseFragment() {

    private lateinit var tabList:ArrayList<CommMap>
    private var contentFragmentPagerAdapter: ContentFragmentPagerAdapter? = null
    private lateinit var tabFragments: Array<Fragment?>

    override fun initData() {
        super.initData()
        tabList = arrayListOf(CommMap("广场",MineOneFragment::class.java.name),
                CommMap("番剧",MineOneFragment::class.java.name),
                CommMap("国创",MineOneFragment::class.java.name),
                CommMap("放映厅",MineOneFragment::class.java.name),
                CommMap("纪录片",MineOneFragment::class.java.name),
                CommMap("漫画",MineOneFragment::class.java.name),
                CommMap("专栏",MineOneFragment::class.java.name),
                CommMap("直播",MineOneFragment::class.java.name),
                CommMap("音频", MusicFragment::class.java.name),
                CommMap("游戏",MineOneFragment::class.java.name))

        tabFragments = kotlin.arrayOfNulls(tabList.size)

        for (i in tabList.indices){
            val clazz = Class.forName(tabList[i].value) as Class<BaseFragment>
            tabFragments[i] = clazz.newInstance()
        }
    }

    override fun initView() {
        super.initView()
        initViewPager()
        initTab()
    }

    override fun initListener() {
        super.initListener()
        di_open_drawer.setOnClickListener(this)
        ci_user_head.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        when(v.id){
            R.id.di_open_drawer,R.id.ci_user_head->{
                (activity as MainActivity).openDrawer()
            }
        }
    }

    private fun initViewPager(){
        contentFragmentPagerAdapter = ContentFragmentPagerAdapter(childFragmentManager, tabList)
        vp_content.adapter = contentFragmentPagerAdapter
        vp_content.currentItem = 0
        vp_content.offscreenPageLimit = tabFragments.size
    }

    private fun initTab() {
        tl_tab.setupWithViewPager(vp_content)
        tl_tab.setTabTextColors(ContextCompat.getColor(_mActivity, R.color.tab_text_color), ContextCompat.getColor(_mActivity, R.color.colorPrimary))
        tl_tab.setSelectedTabIndicatorColor(ContextCompat.getColor(_mActivity,R.color.colorPrimary))
    }
}
